stats/opentelemetry/internal/tracing/custom_carrier.go

```diff
@@ -47,7 +52,18 @@ func NewCustomCarrier(ctx context.Context) *CustomCarrier {
 // Get returns the string value associated with the passed key from the gRPC
 // context. It returns an empty string if the key is not present in the
 // context or if the value associated with the key is empty.
+//
+// If the key is `grpc-trace-bin`, it retrieves the binary value using
+// `c.GetBinary()` and then base64 encodes it before returning. For all other
+// keys, it retrieves the value from the context's metadata.
 func (c *CustomCarrier) Get(key string) string {
+	if key == GRPCTraceBinHeaderKey {
+		return base64.StdEncoding.EncodeToString(c.GetBinary())
+	}
+	if strings.HasSuffix(key, "bin") && key != GRPCTraceBinHeaderKey {
+		logger.Warningf("encountered a binary header %s which is not: %s", key, GRPCTraceBinHeaderKey)
```

Comment:
I think there is some confusion. For example java doesn't have any check to return early for other -bin headers in regular get https://github.com/grpc/grpc-java/pull/11409/files#diff-f7d5bc9bc8a38c2255488580652f3dc247961787da4890b8d00e19e5731ddebbR55.  It will be just treated as a string key. Only in case of getBinary and setBinary that check is there to not allow anything else other than grpc-trace-bin as a -bin header.

Right now in go my SetBInary and GetBinary is little different. It doesn't take any key as they are always going to use stats package which only has one key to get/set grpc-trace-bin header. So, only time they are called is either from GRPCTraceBinPropagator or if regular get/set is called for key grpc-trace-bin. Any other -bin key is treated as regular string key. Basically we will never call binary methods for any other key apart from grpc-trace-bin

Reply:
Discussed offline. Java implementation have a bug. As per design, even regular get/set we shouldn't allow any other binary header except grpc-trace-bin. Modified the code and tests accordingly.